matRad_calcGeoDists.m

```diff
@@ -95,14 +98,18 @@
 ix = radDepthIx(subsetMask);
 
 % return radial distances squared
-if nargout > 1
-    rad_distancesSq = rad_distancesSq(subsetMask);
-end
+rad_distancesSq = rad_distancesSq(subsetMask);
 
 % return x & z distance
-if nargout > 2
-   isoLatDistsX = latDistsX(subsetMask)./rot_coords_temp(subsetMask,2)*SAD;
-   isoLatDistsZ = latDistsZ(subsetMask)./rot_coords_temp(subsetMask,2)*SAD; 
-end
+% if nargout > 2
+%    isoLatDistsX = latDistsX(subsetMask)./rot_coords_temp(subsetMask,2)*SAD;
+%    isoLatDistsZ = latDistsZ(subsetMask)./rot_coords_temp(subsetMask,2)*SAD; 
+% end
+
 
+% latDists
```

Comment:
What I meant is that it probably would make sense to not query latDistX & Z two times when subsetMask doesn't change: 
```
if nargout > 2
  latDistsX = latDistsX(subsetMask);
  latDistsZ = latDistsZ(subsetMask);  
  isoLatDistsX = latDistsX./rot_coords_temp(subsetMask,2)*SAD;
  isoLatDistsZ = latDistsZ./rot_coords_temp(subsetMask,2)*SAD; 
end
```
Meaning you should keep nargouts if it boosts runtime, but just optimize the code for efficiency. But actually I do not know which version is the most efficient, and since calcGeoDists is quite sensitive regarding runtime, could you measure?
Maybe the indexing is faster than reducing the array to isoLatDistsX, or the other way round. if the indexing is faster, it might even make sense to put the latDistX assignment in its own nargout at the end. Sorry for the confusion here.

Reply:
I measured the runtimes of either indexing and initially shrinking the array. Results are that your proposed version is 15% faster for three different example setups. I will implement this method in combniation with the if-statement again.